config-gen/apis/v1alpha1/testdata/componentconfig/expected.yaml

```diff
@@ -0,0 +1,323 @@
+apiVersion: v1
```

Comment:
WDYT about add in the kubebuiler/testdata a sample which would be the project-v3 but with config-gen instead? Then, would not make it easier we compare both scenarios/options to get its advantages and etc?

Reply:
Done

Reply:
(See second commit)

Reply:
I am not sure if it was what was in my mind. I thought in: 
- New testatada project. e.g project-v3-configgen
- where  project-v3-configgen would be == project-v3 without the default config scaffold. I mean, this project would be an example of your end goal idea. Are you not looking for to replace the config/ dir with the config-gen?

Reply:
It is adding the same configuration for all samples. However, each sample has a diff config.
My initial idea was we have a testdata sample to check the config-gen which would be like: 
- a new sample such as testdata/project-v3-config-gen 
- this sample would have the same config of project-v3
- it would be an example of its end-goal idea where the config/ directory was replaced by config-gen

Then, we could compare project-v3 with the new one and ensure that it is covering the same features.
Also, we could use this sample to run the same integration tests and ensure that all would work well as expected. WDYT?

Reply:
I could add that test.  This PR doesn't propose replacing the `config` directory, nor does it have any mechanism for doing so.  Do we want to add that functionality at this time, or defer support for replacing the existing workflow until we are ready to do so?

There are config-gen specific tests in its own package, and I'd prefer we keep the config between project versions consistent to catch any differences between how config is generated differently for them.

Reply:
If I'm allowed to suggest then I think it's better to separate the PR until it's ready.

Reply:
Totally fine with us do that in a follow-up